### PR TITLE
Added example pyproject and package. 

### LINKED ISSUE #3: Add `pyproject.toml` and arrange files per standards
Add a `pyproject.toml` file and create a directory structure that follows typical project layouts for the code, test, etc.  See https://packaging.python.org/en/latest/tutorials/packaging-projects/

### PR DESCRIPTION
**Pull request by MicahGale**
Added an example package with a `pyproject.toml` file.

This a variety of levels of concepts that may or not be appropriate for this scope.

1. a `pyproject.toml` file that covers:
    1. base metadata
    3. dependencies
    4. setuptools/build metadata: `python -m build` and `pip install .` should work on it.
    5. Configuring third-party plugins (`pytest`).
    6. Specifying optional dependencies
1. Packaging covering:
    1. having an `__init__.py`
    3. Importing a class to be top-level from a submodule
    4. Making the module executable via `__main__.py`
    5. Internal package imports
    6. Testing a package
1. OOP:
    1. Using `_private`
    3. Making `@property` getters
    4. Doc strings with sphinx directives
1. Testing:
   1. Structure and imports
   2. `pytest.fixture`
   3. Using `pytest.approx` for float comparisons
1. Amusing myself.

Fixes #3.

### CODE REVIEW

**Comment by gonuke** [comment]
Can we implement this on the files in #1 (and thus merge that ASAP if ready) including moving those files as appropriate?

**Comment by MicahGale** [comment]
Yes, I was thinking this should be based on #1, but I didn't want to deal with a second remote. So I'll make this a draft again until that is merged.

**Comment by gonuke** [comment]
> > Can we implement this on the files in #1 (and thus merge that ASAP if ready) including moving those files as appropriate?
> 
> By this do you mean moving the current files in #1 into the package or not? I was thinking of including some top level files that show good practices for scripting, and then this being an independent-ish folder and showing packaging.

I guess I don't need to get too complicated.  It's meant for people starting new projects from scratch without a lot of prior experience.  Happy to debate what that minimal case looks like, so I'll see how it all fits.  Don't want you to go too far down a rabbit hole though.

**Comment by MicahGale** [comment]
Add example of pyproject scripts/entry points?

**Comment by MicahGale** [comment]
Todo: move `do_task.py` and `test_do_task.py` into `acme_corp`. Harmonize doc strings.

File: acme_corp/__main__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from . import Piano
+
+"""
+This is a special file that makes this package executable.
+
+It will only be ran when this package is ran from the command line, i.e.,
```
Comment (gonuke) [inline acme_corp/__main__.py:6]:
```suggestion
It will only be run when this package is run from the command line, i.e.,
```

File: acme_corp/do_task.py [hunk 2]
```diff
@@ -4,22 +4,20 @@
 
 logger = logging.getLogger(__name__)
 
+# using Sphinx docstrings style: <https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html>
```
Comment (gonuke) [inline acme_corp/do_task.py:7]: did we decide to switch to numpy docstring style?

File: acme_corp/piano.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
+class Piano:
+    """
+    A grand piano proudly built by ACME corporation.
+
+    Weighing half a ton be careful when hoisting this precariously over
+    sidewalks.
+
+    .. warning::
+        Always use proper rigging techniques when hoisting this above
+        above any paths that may be occupied any Road Runners.
```
Comment (gonuke) [inline acme_corp/piano.py:10]:
```suggestion
        above any paths that may be occupied by any Road Runners.
```

File: tests/__init__.py
Comment (gonuke) [inline tests/__init__.py:1]: Is this file necessary?

**Review by gonuke — CHANGES_REQUESTED** [review]
A couple of typos and reminder about docstring format.

File: acme_corp/do_task.py (hunk 2 above)
Comment (MicahGale) [reply acme_corp/do_task.py:7]: I think we decided to make that a separate PR #12.

File: tests/__init__.py
Comment (MicahGale) [reply tests/__init__.py:1]: Pytest isn't happy without it. I can't remember if python has removed the need for this file or not. There might be a workaround, but I'm thinking: "if it ain't broke..."

```
========================================================================= test session starts ==========================================================================
platform linux -- Python 3.10.13, pytest-7.4.0, pluggy-1.3.0
rootdir: /home/mgale/dev/py-template
plugins: anyio-4.0.0, profiling-1.7.0
collected 0 items / 2 errors

================================================================================ ERRORS ================================================================================
________________________________________________________________ ERROR collecting tests/test_do_task.py ________________________________________________________________
ImportError while importing test module '/home/mgale/dev/py-template/tests/test_do_task.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../mambaforge/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_do_task.py:1: in <module>
    import acme_corp.do_task as do_task
E   ModuleNotFoundError: No module named 'acme_corp'
_________________________________________________________________ ERROR collecting tests/test_piano.py _________________________________________________________________
ImportError while importing test module '/home/mgale/dev/py-template/tests/test_piano.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../mambaforge/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_piano.py:1: in <module>
    import acme_corp
E   ModuleNotFoundError: No module named 'acme_corp'
======================================================================= short test summary info ========================================================================
ERROR tests/test_do_task.py
ERROR tests/test_piano.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
========================================================================== 2 errors in 0.11s ===========================================================================
```

**Review by gonuke — APPROVED** [review]
Thanks @MicahGale
